File: backend/Raporlar/GecikenCheckIn/Dto/GecikenCheckInRaporDto.cs
namespace STYS.Raporlar.GecikenCheckIn.Dto;

public class GecikenCheckInRaporDto
{
    public int TesisId { get; set; }
    public string? TesisAdi { get; set; }
    public DateTime ReferansTarihi { get; set; }
    public int? OdaTipiId { get; set; }
    public string? OdaTipiAdi { get; set; }
    public string GecikmeDurumu { get; set; } = "tumu";
    public string Baslik { get; set; } = "";

    public GecikenCheckInOzetDto Ozet { get; set; } = new();
    public List<GecikenCheckInRezervasyonDto> Rezervasyonlar { get; set; } = [];
}

public class GecikenCheckInOzetDto
{
    public int ToplamRezervasyonSayisi { get; set; }
    public int BugunGirisSayisi { get; set; }
    public int GecikenSayisi { get; set; }
    public int KritikGecikenSayisi { get; set; }
    public int ToplamKisiSayisi { get; set; }
    public decimal ToplamKalanTutar { get; set; }
}

public class GecikenCheckInRezervasyonDto
{
    public int RezervasyonId { get; set; }
    public string? ReferansNo { get; set; }
    public string? MisafirAdiSoyadi { get; set; }
    public string? MisafirTelefon { get; set; }

    public DateTime GirisTarihi { get; set; }
    public DateTime CikisTarihi { get; set; }

    public int GecikenGunSayisi { get; set; }
    public int KisiSayisi { get; set; }

    public string RezervasyonDurumu { get; set; } = "";
    public string RezervasyonDurumuLabel { get; set; } = "";

    public string GecikmeDurumu { get; set; } = "";
    public string GecikmeDurumuLabel { get; set; } = "";

    public List<string> OdaNolari { get; set; } = [];
    public List<string> OdaTipleri { get; set; } = [];

    public decimal ToplamUcret { get; set; }
    public decimal OdenenTutar { get; set; }
    public decimal KalanTutar { get; set; }
    public string? ParaBirimi { get; set; }
}

File: frontend/src/app/pages/raporlar/geciken-check-in/geciken-check-in-rapor.dto.ts
export interface GecikenCheckInOzetDto {
    toplamRezervasyonSayisi: number;
    bugunGirisSayisi: number;
    gecikenSayisi: number;
    kritikGecikenSayisi: number;
    toplamKisiSayisi: number;
    toplamKalanTutar: number;
}

export interface GecikenCheckInRezervasyonDto {
    rezervasyonId: number;
    referansNo: string | null;
    misafirAdiSoyadi: string | null;
    misafirTelefon: string | null;

    girisTarihi: string;
    cikisTarihi: string;

    gecikenGunSayisi: number;
    kisiSayisi: number;

    rezervasyonDurumu: string;
    rezervasyonDurumuLabel: string;

    gecikmeDurumu: string;
    gecikmeDurumuLabel: string;

    odaNolari: string[];
    odaTipleri: string[];

    toplamUcret: number;
    odenenTutar: number;
    kalanTutar: number;
    paraBirimi: string | null;
}

export interface GecikenCheckInRaporDto {
    tesisId: number;
    tesisAdi: string | null;
    referansTarihi: string;
    odaTipiId: number | null;
    odaTipiAdi: string | null;
    gecikmeDurumu: string;
    baslik: string;

    ozet: GecikenCheckInOzetDto;
    rezervasyonlar: GecikenCheckInRezervasyonDto[];
}
